src/EventStore.Core/StorageTracker.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using EventStore.Core.TransactionLog.LogRecords;
+
+namespace EventStore.Core;
+
+public class StorageTracker : IStorageTracker {
+	private readonly Counter<long> _readBytes;
+	private readonly Counter<long> _readEvents;
+	private readonly Counter<long> _indexedEvents;
+
+	public StorageTracker(Meter meter) {
+		_readBytes = meter.CreateCounter<long>("ReadBytes", "bytes");
+		_readEvents = meter.CreateCounter<long>("ReadEvents", "events");
+		_indexedEvents = meter.CreateCounter<long>("IndexedEvents", "events");
```

Comment:
open to opinions, should _readEvents and _indexedEvents be dimensions of one instrument? perhaps we should see how the opentelemetry instrumentation libraries break down their time series into instruments/dimensions